tests/integration/test_replicated_database_cluster_groups/test.py

```diff
@@ -0,0 +1,116 @@
+import re
+import pytest
+
+from helpers.cluster import ClickHouseCluster
+from helpers.test_tools import assert_eq_with_retry
+
+cluster = ClickHouseCluster(__file__)
+
+main_node_1 = cluster.add_instance(
+    "main_node_1",
+    user_configs=["configs/settings.xml"],
+    with_zookeeper=True,
+    stay_alive=True,
+    macros={"shard": 1, "replica": 1},
+)
+main_node_2 = cluster.add_instance(
+    "main_node_2",
+    user_configs=["configs/settings.xml"],
+    with_zookeeper=True,
+    stay_alive=True,
+    macros={"shard": 1, "replica": 2},
+)
+backup_node_1 = cluster.add_instance(
+    "backup_node_1",
+    main_configs=["configs/backup_group.xml"],
+    user_configs=["configs/settings.xml"],
+    with_zookeeper=True,
+    stay_alive=True,
+    macros={"shard": 1, "replica": 3},
+)
+backup_node_2 = cluster.add_instance(
+    "backup_node_2",
+    main_configs=["configs/backup_group.xml"],
+    user_configs=["configs/settings.xml"],
+    with_zookeeper=True,
+    stay_alive=True,
+    macros={"shard": 1, "replica": 4},
```

Comment:
Let's try 1 and 2 as replica names. AFAIR, we use `/<db_path>/log/query-xxxxx/finished/<full_replica_name>` to mark that the query was executed by a specific host. So we should either forbid using the same names in different groups, or add the group name to full replica name (aka host_id)

Reply:
I changed the implementation and tried to allow the same names for the replicas. Still, in RMT/SMT we use `.../<shard_name>/replicas/<replica_name>` path for replicas, without checks for `|` in the replica name, so we would need to move the replicas somewhere else to avoid a backward incompatibility. So it kind of looks like a lot of work without any real advantages, so for now, I just forbade the creation of replicas with the same name.